src/main/steps/applicant/support-you-need-during-case/SupportYouNeedDuringCaseController.ts

```diff
@@ -85,4 +50,46 @@ export class SupportYouNeedDuringYourCaseController extends PostController<AnyOb
       throw new Error('SafetyConcernsPostController - Case could not be updated.');
     }
   }
+
+  private getSupportYouNeedForApplicant(req: AppRequest<Partial<Case>>) {
+    if ('C100' === req.session.userCase.caseTypeOfApplication) {
+      req.session.userCase?.applicants?.forEach((applicant: Applicant) => {
+        if (applicant?.value?.user?.idamId === req.session?.user.id) {
+          if (req.url.includes('support-you-need-during-case')) {
+            applicant.value.response = {
+              ...applicant.value.response,
+              ...setSupportDetails(req),
+            };
+          }
+        }
+      });
+    } else {
+      if (req.url.includes('support-you-need-during-case')) {
+        req.session.userCase.applicantsFL401!.response = {
+          ...req.session.userCase.applicantsFL401!.response,
+          ...setSupportDetails(req),
+        };
+      }
+    }
+  }
+
+  private getSupportYouNeedForRespondent(req: AppRequest<Partial<Case>>) {
+    if ('C100' === req.session.userCase.caseTypeOfApplication) {
+      req.session.userCase?.respondents?.forEach((respondent: Respondent) => {
```

Comment:
reduce run time complexity

Reply:
The same case with respondents also... as that C100 have multiple respondents... we can have forEach or for-of loops for that respondent array.